### PR TITLE
Fix vm state bug and deletion bug

### LINKED ISSUE #337: machine status still showing 'failed' after restart a 'stopped' instance
**What happened**:
Without MHC for MachineDeployment, when I stopped the compute instance, machine object in 'failed' status as expected and worker node in "NotReady" status as expected. While after restarting the compute instance, the worker node back "Ready" status, while machine object still in "failed" status, capoci log showing the ocimachine already 'Active'
**What you expected to happen**:
machine object should be in "Running" status
**How to reproduce it (as minimally and precisely as possible)**:
deploy a MD without MHC, and stop the worker node, after the machine object turned to "failed", then start the worker node.
**Anything else we need to know?**:

**Environment:**

- CAPOCI version: v0.11.2
- Cluster-API version (use `clusterctl version`): v1.4.1
- Kubernetes version (use `kubectl version`): 1.24.8
- Docker version (use `docker info`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
**What this PR does / why we need it**:
Fix vm state bug and deletion bug

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #337 , #339

### CODE REVIEW

**Comment by shyamradhakrishnan** [comment]
e2e
```
nchronizedAfterSuite] PASSED [0.000 seconds]
[SynchronizedAfterSuite]
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/e2e_suite_test.go:261
------------------------------
[SynchronizedAfterSuite] PASSED [597.206 seconds]
[SynchronizedAfterSuite]
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/e2e_suite_test.go:261

  Timeline >>
  STEP: Tearing down the management cluster @ 10/19/23 06:24:06.696
  << Timeline
------------------------------

Ran 7 of 28 Specs in 2999.034 seconds
SUCCESS! -- 7 Passed | 0 Failed | 0 Pending | 21 Skipped

```

**Comment by shyamradhakrishnan** [comment]
unit test
```
$ make test
GOBIN=/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/hack/tools/bin ./scripts/go_install.sh sigs.k8s.io/controller-runtime/tools/setup-envtest setup-envtest v0.0.0-20230131074648-f5014c077fc3
kube-builder assets: /home/ubuntu/.local/share/kubebuilder-envtest/k8s/1.24.2-linux-amd64
KUBEBUILDER_ASSETS="/home/ubuntu/.local/share/kubebuilder-envtest/k8s/1.24.2-linux-amd64" go test -coverprofile=coverage.out ./...
?   	github.com/oracle/cluster-api-provider-oci	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta1	51.774s	coverage: 23.7% of statements
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta2	0.039s	coverage: 15.6% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/config	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/metrics	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/cloud/ociutil	0.005s	coverage: 19.6% of statements
ok  	github.com/oracle/cluster-api-provider-oci/cloud/scope	231.084s	coverage: 75.1% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base/mock_base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement/mock_computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine/mock_containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/loadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/loadbalancer/mock_lb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/cloud/util	0.212s	coverage: 61.1% of statements
ok  	github.com/oracle/cluster-api-provider-oci/controllers	26.736s	coverage: 58.6% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta1	14.028s	coverage: 15.3% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta2	0.016s	coverage: 4.8% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/controllers	0.791s	coverage: 56.3% of statements
?   	github.com/oracle/cluster-api-provider-oci/feature	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/version	[no test files]
go tool cover -func=coverage.out -o coverage.txt
go tool cover -html=coverage.out -o coverage.html
```

File: controllers/ocimachine_controller.go [hunk 1]
```diff
@@ -326,6 +326,10 @@ func (r *OCIMachineReconciler) reconcileNormal(ctx context.Context, logger logr.
 		machineScope.Info("Instance is pending")
 		conditions.MarkFalse(machineScope.OCIMachine, infrastructurev1beta2.InstanceReadyCondition, infrastructurev1beta2.InstanceNotReadyReason, clusterv1.ConditionSeverityInfo, "")
 		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	case core.InstanceLifecycleStateStopping, core.InstanceLifecycleStateStopped:
+		machineScope.SetNotReady()
```
Comment (joekr) [inline controllers/ocimachine_controller.go:330]: Do we want to flag this same state for Terminating, Terminated?

**Review by joekr — DISMISSED** [review]

File: controllers/ocimachine_controller.go (hunk 1 above)
Comment (shyamradhakrishnan) [reply controllers/ocimachine_controller.go:330]: good point, done.please review

**Review by joekr — APPROVED** [review]

**Pull request merged**